christfellowshipchurch/web-app-v2
# Cleanup behavior in Group Search All Filters modal

## About
Handful of mostly behavioral changes to the Group Finder search results page and the All Filters modal, and some odds-and-ends along the way:

- 🐞 Fix bug with "Clear All" button
- Refine filters state hydration to allow empty/blank/wide-open searches (no initial URL search params)
- Prevent searches awkwardly executing behind the modal while changing filters
- Rename "Apply" button to "Done", since it was misleading (changes are technically applied immediately)
- Don't pre-populate campus filter with users' campus when opening All filters modal from search results
- Update placeholder text for text search to "Search for a group..."
- Tweak margin between `<FilterField>` components
- Update the `preference` prop on `<GroupCard>` component to be `preferences`, and display multiple values if present

## Testing
Hammer on the group search filters flow:
- Open and close the Filters modal without making changes
- Ensure the group filter buttons are hidden when the modal is open
- Ensure Clear All works
- Ensure searches do not execute in the background while modal is open
- Ensure wide-open blank searches work (land on `/community/search` directly, no search params)

## Screenshots

### Most of the changes
![2021-02-25 13 45 02](https://user-images.githubusercontent.com/1812955/109201588-05a9f900-7770-11eb-85fa-faaeb7452a58.gif)

### `GroupCard` component Storybook
![CleanShot 2021-02-25 at 11 26 48@2x](https://user-images.githubusercontent.com/1812955/109203336-09d71600-7772-11eb-8adc-0619eea2615d.png)
